exporter/otlphttpexporter/otlp.go

```diff
@@ -259,47 +300,74 @@ func handlePartialSuccessResponse(resp *http.Response, partialSuccessHandler par
 
 type partialSuccessHandler func(bytes []byte, contentType string) error
 
-func tracesPartialSuccessHandler(protoBytes []byte, contentType string) error {
-	if contentType != protobufContentType {
+func tracesPartialSuccessHandler(responseBytes []byte, contentType string) error {
+	if contentType != protobufContentType && contentType != jsonContentType {
 		return nil
 	}
 	exportResponse := ptraceotlp.NewExportResponse()
-	err := exportResponse.UnmarshalProto(protoBytes)
-	if err != nil {
-		return fmt.Errorf("error parsing protobuf response: %w", err)
+
+	if contentType == protobufContentType {
+		err := exportResponse.UnmarshalProto(responseBytes)
+		if err != nil {
+			return fmt.Errorf("error parsing protobuf response: %w", err)
+		}
+	} else if contentType == jsonContentType {
+		err := exportResponse.UnmarshalJSON(responseBytes)
+		if err != nil {
+			return fmt.Errorf("error parsing json response: %w", err)
+		}
 	}
+
 	partialSuccess := exportResponse.PartialSuccess()
 	if !(partialSuccess.ErrorMessage() == "" && partialSuccess.RejectedSpans() == 0) {
 		return consumererror.NewPermanent(fmt.Errorf("OTLP partial success: %s (%d rejected)", partialSuccess.ErrorMessage(), partialSuccess.RejectedSpans()))
 	}
 	return nil
 }
 
-func metricsPartialSuccessHandler(protoBytes []byte, contentType string) error {
-	if contentType != protobufContentType {
+func metricsPartialSuccessHandler(responseBytes []byte, contentType string) error {
+	if contentType != protobufContentType && contentType != jsonContentType {
 		return nil
 	}
 	exportResponse := pmetricotlp.NewExportResponse()
-	err := exportResponse.UnmarshalProto(protoBytes)
-	if err != nil {
-		return fmt.Errorf("error parsing protobuf response: %w", err)
+
+	if contentType == protobufContentType {
+		err := exportResponse.UnmarshalProto(responseBytes)
+		if err != nil {
+			return fmt.Errorf("error parsing protobuf response: %w", err)
+		}
+	} else if contentType == jsonContentType {
+		err := exportResponse.UnmarshalJSON(responseBytes)
+		if err != nil {
+			return fmt.Errorf("error parsing json response: %w", err)
+		}
```

Comment:
I think this could also be a switch statement that errors if one of the expected content types is not received.

Reply:
i changed to a switch statement but left previous behaviour, before this PR if the content-type was not proto a `nil` was returned